src/arch/cortex-m4/syscalls.S

```diff
@@ -13,36 +13,28 @@
 
 .thumb_func
 __wait:
-    push {r4-r11,lr}
+    push {lr}
     svc 0
-    pop {r4-r11,lr}
+    pop {lr}
```

Comment:
arm trick: you can pop the link register directly to the pc to save an instruction, i.e.:

```
pop {lr}
bx lr

and

pop {pc}
```
are equivalent.